arch/arm/core/aarch64/switch.S

```diff
@@ -48,6 +48,19 @@ SECTION_FUNC(TEXT, z_arm64_context_switch)
 	mov	x1, sp
 	str	x1, [x2]
 
+#ifdef CONFIG_THREAD_LOCAL_STORAGE
+	/* Grab the TLS pointer */
+	ldr	x2, =_thread_offset_to_tls
+	add	x2, x2, x0
+	ldr	x2, [x2]
+
+	/* Store in the "Thread ID" register.
+	 * This register is used as a base pointer to all
+	 * thread variables with offsets added by toolchain.
+	 */
+	msr	tpidr_el0, x2
```

Comment:
Please note that there is already a WiP about this: see https://github.com/zephyrproject-rtos/zephyr/pull/26545/files#diff-c09baf9cbf3976080a45358981ea144fR70

Reply:
Thanks. Last time I checked there was no activity since August, and I didn't know it has been updated recently.

Reply:
@carlocaione I asked Daniel to pull in the TLS stuff from that PR, which is a larger effort to enable picolibc. Depending on which goes in first we can rebase.

Reply:
@keith-packard  FYI

Reply:
Thanks much, I've read through the patch series and made some comments. Overall, its  clear that this PR should be merged before the picolibc one as picolibc requires TLS support when picolibc is built to use TLS.